spec.mediawiki

```diff
@@ -272,70 +268,48 @@ matching epoch index, '''''i''''', and epoch range are computed as
-epoch. Market orders will only match with "standing" limit orders from previous
-epoch. Limit orders match best-price-first. If more than one standing limit 
-order has the same price, they will be shuffled before matching. 
-
-Orders accumulated in the epoch are matched with standing book orders, where a 
-valid order pair may be:
-
-* a market buy and a limit sell
-* a limit buy and a market sell
-* a limit buy and a limit sell with overlapping limit prices
-
-Each order in the epoch order pool is assigned a priority by shuffling the order 
-pool.
-Shuffling will be deterministic, using the Fisher-Yates algorithm where the 
+First, the order pool is shuffled.
+Shuffling is deterministic, using the Fisher-Yates algorithm where the 
 random number generator seed is derived from the hash of the concatenated order 
-ID hashes.
+[[#client-order-management|ID hashes]].
 Specifically for hash function '''''f''''', the seed hash, 
 '''''H<sub>seed</sub>''''' is computed as 
 
 <!--H_{seed} = f(f(order_1) || f(order_2) || ... || f(order_N))-->
 [[File:spec/images/seed-eq.png]]
 
 where '''''order<sub>i</sub>''''' represents the order ID of the ith order in a 
-lexicographically sorted list of order IDs, '''''||''''' indicates 
-concatenation, and '''''/''''' is integer division.
+lexicographically sorted list of order IDs and '''''||''''' indicates 
+concatenation.
 
-Each order's unique ID and corresponding hash are assigned as the order is 
-placed, and published at the start of the matching process.
+The order IDs and final hash are published at the start of the matching process.
 This is done so that the shuffling may be independently computed for external 
 verification and to facilitate server mesh operation.
-Section 5 describes the order submission and ID assignment.
 
-Cancellations are processed first. Cancellation can potentially carry a penalty 
-if the client's cancel:complete ratio falls below community conduct guidelines. 
+Orders are processed one at a time. Each order is matched according to its type.
 
-Having prioritized each taker order in the epoch pool with a pseudorandom and 
-verifiably unbiased process, matching begins with the first order in the list.
-The order is paired with the best possible 
-(in terms of rate) compatible standing limit order(s), and the orders are queued 
-for monitoring by the DEX. Both clients are immediately informed.
+1. If this current order is a '''cancel order''', any corresponding standing limit order is removed from the list and the cancel order is considered filled. If a cancellation is processed before the order that it cancels, the cancellation will fail, and will need to be re-submitted.
 
-The process continues with the next order in the list and iterates until all 
-orders are removed from the list.
+2. If the order is a limit order than cannot immediately match ('''a maker'''), it is added to the standing orders. It is immediately able to match orders further down in the queue.
```

Comment:
Except in the case of a limit order with time in force "immediate" (or whatever we want to call it), which will not be added to the standing orders, in which case the order is (canceled might be confusing here) terminated without a fill.